### PR TITLE
Fix SM100 blk64 BSA partial KV tails without padding

### LINKED ISSUE #859: SM100 blk64 BSA partial KV tails can produce address-sensitive non-finite output
### Description

The SM100 block-sparse attention forward path with `sparse_block_size=64` can return non-finite output when K/V has a partial final block. Two independently reproducible symptoms appear to share this condition:

1. The BSHD frontend path intermittently returns non-finite output for fixed, finite Q/K/V and unchanged sparse-route metadata.
2. Native contiguous BHSD becomes address-sensitive: changing only a live padding allocation can make the same numerical inputs and route return non-finite output.

The captured shape is `(B=1, H=7, S=37719, D=128)`, so the final KV block contains 23 valid rows. All failing query blocks select that final block.

### Steps to reproduce

Run SM100 blk64 BSA with BF16 Q/K/V, `kv_splits=1`, `use_clc=False`, `allow_empty_block_nums=False`, and `softmax_scale=128**-0.5`.

- Repeating the BSHD call 20 times produced 9 non-finite outputs.
- Repeating native BHSD while varying an otherwise unused live CUDA allocation reproduced at iteration 132 (`padding_mib=1435`, `nonfinite=121167`) in two fresh processes.
- Repeating the same native BHSD call without changing allocation state produced 1000/1000 finite, bit-identical outputs.

The inputs and sparse-route metadata remain unchanged and finite throughout.

### Expected behavior

The output remains finite and independent of unrelated allocation addresses for valid partial-tail inputs.

### Actual behavior

Some allocation/layout states return non-finite output. Physically zero-padding K/V to the next 64-token boundary, while preserving the original final `block_sizes` value, eliminates the failure. The unpadded exact-layout result matches that padded reference bit-for-bit in the captured case.

### Environment

- GPU: NVIDIA B300 SXM6 AC (SM100)
- Driver: 580.178.04
- VBIOS: 97.10.64.00.0C
- PyTorch: 2.10.0+cu130
- CUDA reported by PyTorch: 13.0
- cuDNN Frontend Python source: 1.27.0
- NVIDIA CUTLASS DSL: 4.5.3
- Python: 3.12

Both captures contain only finite BF16 tensors and integer route metadata; no model data is involved.

### PR DESCRIPTION
**Pull request by yujincheng08**
Located by Humanize https://humanfia.ai/ and fixed via KDA https://github.com/mit-han-lab/kernel-design-agents

## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [x] I reviewed the Hard Rules in the `AGENTS.md` for each directory this PR touches and the changes comply.
- [x] Maintainer action: add `cat-bugfix`, `area:sparse_attention`, and `orig-nv-eng` (the PR author does not have label permissions).

## Affected area

FE OSS kernels or CuTeDSL

## Summary

Use an exact-boundary rank-6 K/V TMA view when the SM100 blk64 BSA input has a partial 64-token tail. Coalesce the tiled outer coordinate so TMA issue follows the existing packed fast path.

Add regression coverage for both BHSD and BSHD inputs with a one-token final block.

## Why

The packed K/V view rounds the sequence extent up to a complete 64-token block. TMA therefore considers rows beyond the physical allocation in-bounds, and address-sensitive data from those rows can produce non-finite output.

Keeping the true sequence extent lets TMA zero-fill the out-of-bounds rows. The extra logical mode preserves the efficient descriptor form, and coalescing the outer tile coordinate avoids the coordinate-expansion overhead of the prior exact-layout fallback.

KDA/IKET in-kernel profiling was used to isolate the overhead to the TMA issue path and verify that coalescing removed the coordinate-expansion cost. The profiling instrumentation is not included in the submitted code.

## Related issues

Fixes #859

## API and compatibility impact

No public API change. Aligned K/V lengths keep the existing path. On an NVIDIA B300, an uninstrumented 500-iteration comparison measured 1.973 ms median for the unpadded fixed path versus 1.983 ms for the physically padded fast-path reference; no performance regression was observed.

## Testing

- `python -m pytest -q test/python/fe_api/bsa/test_BSA_attention_forward.py::test_bsa_attention_forward_sm100_blk64_partial_kv_tail_matches_reference`: passed on B300.
- `python -m pytest -q test/python/fe_api/bsa/test_BSA_attention_forward.py::test_bsa_attention_forward_sm100_blk64`: passed on B300.
- Address-perturbation soak on the captured `B=1, H=7, Sk=37719, D=128` case: 1024/1024 finite for contiguous BHSD and 1024/1024 finite for non-contiguous BSHD.
- Split-KV soak: 128/128 finite with `kv_splits=2`.
- CLC soak: 128/128 finite with `use_clc=True`.
- Unpadded output was bit-identical to an explicitly zero-padded K/V reference.
- KDA/IKET traces verified the optimized exact-boundary path retained the packed TMA issue behavior; final performance numbers were collected without instrumentation.
- `pre-commit run --files python/cudnn/block_sparse_attention/_interface.py python/cudnn/block_sparse_attention/csrc/fwd/sm100_blk64/bsa_fwd_sm100.py test/python/fe_api/bsa/test_BSA_attention_forward.py`: passed.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved block-sparse attention handling for KV sequences whose lengths are not multiples of 64.
  - Added support for large KV strides in applicable attention layouts.
  - Ensured partial KV tiles are handled safely with zero-filled values.

- **Tests**
  - Added coverage for partial KV-tail detection on supported GPU architectures.
  - Verified outputs against reference results and across supported tensor layouts.

### CODE REVIEW

File: test/python/fe_api/bsa/test_BSA_attention_forward.py [hunk 1]
```diff
@@ -252,6 +252,70 @@ def test_bsa_attention_forward_sm100_blk64():
     assert split_result_bshd["o_tensor"].stride() == result_bshd["o_tensor"].stride()
 
 
+@pytest.mark.L0
+@pytest.mark.parametrize(("seqlen_k", "expected"), [(4 * 64, False), (4 * 64 - 1, True)])
+def test_bsa_attention_forward_sm100_blk64_detects_partial_kv_tail(seqlen_k, expected):
+    interface = importlib.import_module("cudnn.block_sparse_attention._interface")
```
[the remark this answers is not in the document]
Comment (yujincheng08) [reply test/python/fe_api/bsa/test_BSA_attention_forward.py:259]: Fixed in 0d2f1a8 by calling `_import_bsa()` before importing the interface, preserving the optional-dependency skip behavior. The targeted pre-commit checks pass.

**Comment by jiayus-nvidia** [comment]
@cudnn-ci-bot run oss

**Review by jiayus-nvidia — APPROVED** [review]
